Examples/test-suite/cpp11_attribute_specifiers.i

```diff
@@ -41,15 +50,33 @@ bool maybeUnused2(bool a, [[maybe_unused]] bool b) { return a; }
   return false;
 }
 
-struct [[nodiscard]] S { };
+enum E {
+  oldval [[deprecated("Replaced by newval")]],
+  newval
+};
+
+[[gnu::aligned(16)]] int aligned_var_1 = 0;
+int aligned_var_2 [[gnu::aligned(__BIGGEST_ALIGNMENT__)]] = 0;
+
+[[gnu::alloc_size (1, 2)]] typedef void* (*calloc_ptr) (size_t, size_t);
+struct [[nodiscard, gnu::aligned(16)]] S {
+  [[gnu::visibility("default")]] virtual int visible(int a) { return a + 1; }
+  [[gnu::visibility("hidden")]] int hiddenWithString(const int &n = int()) { return 4; }
+  [[gnu::visibility("hidden")]] static int hidden(int a) { return a + 2; }
+  int x;
+  virtual ~S() = default;
+};
 
 const char *test_string_literal() { return "Test [[ and ]] in string literal"; }
 
-#if 0
+[[gnu::visibility("default")]] int visible(int a) { return a + 1; }
+[[gnu::visibility("hidden"), gnu::always_inline]] int hidden(int a) { return a + 2; }
+
 // Check that SWIG doesn't choke on ]] when it's not part of an attribute.
-// FIXME: SWIG's parser doesn't handle this case currently.
-int *a;
-int b = a[a[0]];
-#endif
+void test_fn() {
+  int data[1] = { 0 };
+  int *a = data;
+  int b = a[a[0]];
+}
```

Comment:
The point here is to ensure that SWIG can parse `int b = a[a[0]];` (in particular that the `]]` is broken apart properly.  SWIG doesn't parse function bodies, so mobing this code into a function body doesn't actually test what was intended here.

(The reason this part is currently deactivated is just that the parser doesn't currently actually handle a double array dereference in an expression, but once that is fixed we want to be testing this properly.)

Reply:
It will still have to be tokenized if attributes parsing is disabled - this is the only case when this could not work.

Reply:
The parser just skips tokens between the `{` at the start of a function body and its matching closing `}`, so putting it in a function body will parse OK even if the `]]` is scanned as a single token.

Reply:
The next time you permit yourself to make a comment in order to pass a criminal message, I will seriously reconsider my decision from last year to skip sending an email to the legal department of every institution mentioned in the SWIG license files to inform them of a copyright problem involving plagiarism for a criminal extortion related to a series of falsified legal procedures in the EU. I will be completely honest and I will attest that you most probably do not have anything to do with the organized judicial and police corruption, the sexual elements, the international prostitution ring or the drug trade taking part in this affair. My patience has limits.

Reply:
@mmomtchev, this last comment appears threatening, is not constructive and is not welcome in the collaborative SWIG development community that we strive for. @ojwb is merely trying to help review and improve your (very welcome) contribution. I don't understand the comment as it appears out of context and I hope it has been posted on the wrong issue, in which case, please withdraw it. If not and if you have some non-technical issue about SWIG that you would like to discuss, please raise a separate discussion thread. Otherwise, I ask you to kindly keep discussions on topic to C++11 attributes. Thank-you for you understanding.